config/http.conf

```diff
@@ -9,15 +9,36 @@ http {
   client_max_body_size  400m;
   client_body_timeout   300s;
 
+  # If we receive X-Forwarded-Proto, pass it through; otherwise, pass along the
+  # scheme used to connect to this server
+  map $http_x_forwarded_proto $proxy_x_forwarded_proto {
+    default $http_x_forwarded_proto;
+    ''      $scheme;
+  }
+
+  # If we receive X-Forwarded-Proto, use it to determine the value of
+  # X-Forwarded-Ssl
+  map $http_x_forwarded_proto $proxy_x_forwarded_ssl {
+    default off;
+    https   on;
+  }
```

Comment:
Basically this means, if the value of `$http_x_forwarded_proto` is `https` then set a variable `$proxy_x_forwarded_ssl` to `on` ... and since that's the only other option in the `map` then if we don't match `https` then default to `default` which is defined as `off`.

Real world example: if using AWS LBs, `X-Forwarded-Ssl` **will NOT be set by the LB** and therefore `$http_x_forwarded_ssl` is not available as . See [classic-ELB](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html) and [ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/x-forwarded-headers.html) docs on the `X-Forwarded-*` headers. Therefore we rely on the `X-Forwarded-Proto` header which will be set to either `http` or `https` as explained in [PLACEHOLDER].